server/api/src/rsGeneralAdmin.cpp

```diff
@@ -52,6 +53,63 @@ namespace
         return std::find_if(b, e, [](unsigned char _ch) { return ::isspace(_ch); }) != e;
     } // contains_whitespace
 
+     auto throw_if_inappropriate_rodsadmin_downgrade(
+        RsComm& rsComm,
+        const std::string_view& _new_value,
+        const std::string_view& _user_name
+    ) -> void {
+            std::set<std::string> server_administrators_in_zone;
+
+            BytesBuf* bbuf = nullptr;
+            irods::at_scope_exit free_buf { [&bbuf] { std::free(bbuf); }  };
+
+            if (const auto ec = rsZoneReport(&rsComm, &bbuf); ec < 0 ) {
+                const auto msg = fmt::format("[{}-{}] - Error obtaining zone report", __func__, __LINE__);
+                addRErrorMsg(&rsComm.rError, ec, msg.c_str());
+                THROW(ec, msg);
+            }
+
+            nlohmann::json zone_report = nlohmann::json::parse(
+                    std::string(static_cast<char *>(bbuf->buf), bbuf->len)
+            )["zones"];
+
+            const std::string local_zone_name = std::string(getLocalZoneName());
+
+            for ( const auto& zone : zone_report ) {
+                if ( zone["icat_server"]["service_account_environment"]["irods_zone_name"] == local_zone_name ) {
+
+                    server_administrators_in_zone.insert(
+                        static_cast<std::string>(zone["icat_server"]["service_account_environment"]["irods_user_name"])
+                    );
+
+                    for ( const auto& server : zone["servers"] ) {
+                        server_administrators_in_zone.insert(
+                            static_cast<std::string>(server["service_account_environment"]["irods_user_name"])
+                        );
+                    }
+
+                }
+            }
+
+            if ( server_administrators_in_zone.find(static_cast<std::string>(_user_name)) != server_administrators_in_zone.end() &&
+                    _new_value != "rodsadmin"
+                )
+            {
+                const auto msg = fmt::format("[{}-{}] - Cannot downgrade another rodsadmin running another server in this zone.", __func__, __LINE__);
+                addRErrorMsg(&rsComm.rError, SYS_NOT_ALLOWED, msg.c_str());
+                THROW(SYS_NOT_ALLOWED, msg);
+            }
+
+            if ( std::string(rsComm.clientUser.userName) == _user_name &&
```

Comment:
`std::string_view` supports `operator==()` for string literals, so the use of `std::string` is unnecessary.
```cpp
if (rsComm.clientUser.userName == _user_name && _new_value != "rodsadmin") {
```